lib/src/local_working_copy.rs

```diff
@@ -335,6 +336,7 @@ pub struct TreeState {
     sparse_patterns: Vec<RepoPathBuf>,
     own_mtime: MillisSinceEpoch,
     symlink_support: bool,
+    conflict_marker_style: ConflictMarkerStyle,
```

Comment:
Perhaps, this can be passed as `CheckOutOptions { conflict_marker_style }`?

It's a bit weird that we have to pass the same value to `snapshot()` (to reconstruct the old file content), but I think it can be simplified later. For example, tree state could remember a hash of conflict file content for comparison.

Reply:
The plumbing code looks good to me, thanks.